.github/workflows/pre-commit.yml

```diff
@@ -1,10 +1,14 @@
 name: Code formatting
 
 # see: https://help.github.com/en/actions/reference/events-that-trigger-workflows
-on:  # Trigger the workflow on push or pull request, but only for the main branch
-  push:
-    branches: [main]
-  pull_request: {}
+on:  # Trigger the workflow on pull request or merge
+  pull_request:
+    paths:
+      - '.pre-commit-config.yaml'
+      - 'autogen/**'
+      - 'samples/**'
```

Comment:
What's the reason to specify `paths`? Is it because you'd like to exclude the `dotnet`? The current spec excludes more than that. Also, the `merge_group` trigger does not have this spec. Will there be a problem?

Reply:
> What's the reason to specify paths? Is it because you'd like to exclude the dotnet?

The path filter is added so the workflow will be triggered only when there’s change in python code or code format configuration. That seems to be what this workflow suppose to do? Do we want to run that workflow under other circumstances?

Reply:
@sonichi Maybe a better way is to set up `exclude` path filter in `pre-commit-config.yaml`?